src/window-system/x11-cairo-surface.cpp

```diff
@@ -57,6 +57,14 @@ X11CairoSurface::X11CairoSurface(Display *display) : display(display) {
       CWColormap | CWBorderPixel | CWBackPixel | CWOverrideRedirect, &attr);
   XMapWindow(display, this->window);
 
+  // Add WM_NAME and WM_CLASS properties to the window
+  std::string name("touchégg");
+  XStoreName(display, this->window, &name[0]);
+  XClassHint *classHint = XAllocClassHint();
+  classHint->res_name = &name[0];
+  classHint->res_class = &name[0];
+  XSetClassHint(display, this->window, classHint);
```

Comment:
Don't forget to free memory:

```suggestion
  XSetClassHint(display, this->window, classHint);
  XFree(classHint);
```